app/models/openscap_result.rb

```diff
@@ -0,0 +1,76 @@
+class OpenscapResult < ApplicationRecord
+  belongs_to :container_image
+  has_one    :binary_blob,           :dependent => :destroy, :autosave => true, :as => :resource
+  has_many   :openscap_rule_results, :dependent => :destroy, :autosave => true
+
+  def self.openscap_available?
+    # needed only for travis
+    require 'openscap'
+    require 'openscap/ds/arf'
+    require 'openscap/xccdf/benchmark'
+    true
+  rescue LoadError
+    false
+  end
+
+  def raw=(value)
+    raise "Cannot invoke raw more than once." unless binary_blob.nil?
+    self.binary_blob = BinaryBlob.new(:name => 'openscap_compliance_arf', :data_type => 'XML')
+    binary_blob.binary = value
+    with_openscap_objects(binary_blob.binary) do |rule_results, benchmark_items|
+      create_results(rule_results, benchmark_items)
+    end
+  end
+
+  def html # Must be called after setting raw or it will explode
```

Comment:
Is it wrong to want it to explode if there is no binary_blob?
I do not want to handle a returned '' and it should never happen (after adding :required => true as you suggested I think It can only happen in an object not yet saved)